src/main/java/backend/Logic.java

```diff
@@ -206,14 +201,41 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        boolean isLabelsChanged = updateIssueLabels(issue, newLabels);
+
+        if (isLabelsChanged) {
+            logger.info("Changing labels for " + issue + " on UI");
+            refreshUI();
+        }
+
+        return updateIssueLabelsOnRepo(issue, newLabels);
+    }
```

Comment:
You missed this comment:
Why update on the repo even if there is no change in labels? If there is a reason, it should be given as a comment.